### PR TITLE
docs: a listing component

### LINKED ISSUE #37: Ibimina - Single Listing Component
### Single Listing Component
This comprises of the listing image/logo, listing name, listing description, manager name, a link to a detailed page for the listing

### PR DESCRIPTION
**Pull request by ibimina**
## What type of PR is this? (check all applicable)

- [ ] Feature
- [ ] Refactor
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update
- [x] Other (please specify here: Component creation)
  
  
## Description of what this Pull Request does
(type your description here)
A listing component in my application that can display data passed as props. This would allow easy creation of dynamic lists of items in the landing page and customize the content and appearance of the list based on the data provided.

- The listing component should accept a data prop which allows developers to pass data to be displayed in the list.
- The listing component should be able to render the list items accordingly.
- The listing component should display the data passed via the data prop in the format specified by the component's implementation 

## Related Tickets & Documents
- Closes #37 
- Related Issues: (enter ticket numbers here)
  
## QA Instructions, Screenshots, Recordings

  listing landing page

update listing

## Added/updated tests?
- [ ] Yes
- [ ] Not Yet, and here is why: __explain here__
- [ ] I need help with tests
- [x] No, and here is why: __explain here__

### CODE REVIEW

**Comment by jjleigh** [comment]
Can you add a screenshot of what the component looks like in the UI

File: src/components/landing/listing.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import Link from "next/link";
+import Image from "next/image";
+
+function Listing({ data }) {
+  return (
+    <>
+      <article className="listing">
+        <div className="listing__image">
```
Comment (annonymousauthority) [inline src/components/landing/listing.js:8]: Wouldn't this class style be confused with the image style itself?

Comment (ibimina) [reply src/components/landing/listing.js]: I have change the class Name from "listing__image" to "listing__image__container"

**Comment by ibimina** [comment]
> > Can you add a screenshot of what the component looks like in the UI
> 
> Will attached a screeenshot
@jjleigh 
listing landing page

File: src/styles/globals.css [hunk 2]
```diff
@@ -1,107 +1,14 @@
...
-@media (prefers-color-scheme: dark) {
-  html {
-    color-scheme: dark;
-  }
-}
+  font-family: 'Roboto', sans-serif;
+  font-size: 16px;
+  width: 100%;
+  min-height: 100vh;
+}
```
Comment (jjleigh) [inline src/styles/globals.css:14]: Add a new line at the end of the file.

File: src/styles/Home.module.css [hunk 3]
```diff
@@ -1,278 +1,17 @@
...
+         grid-template-columns: repeat(2, 1fr);
+         gap: 1em;
+     }
+ }
+
+ @media (min-width: 60rem) {
+     .main {
+         grid-template-columns: repeat(3, 1fr);
+     }
+ }
```
Comment (jjleigh) [inline src/styles/Home.module.css:17]: Add a new line at the end of the file.

File: src/components/landing/listing.module.css [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+    gap: 1em;
+    border: 1px solid #bab7b7;
+    border-radius: 5px;
+    padding: 1em;
+}
+
+.listing__view__more {
+    text-align: end;
+    display: block;
+}
```
Comment (jjleigh) [inline src/components/landing/listing.module.css:12]: Add a new line here at the end

**Review by jjleigh — COMMENTED** [review]
I see you have attached a screenshot in a quoted reply, but can you add it to the description of the PR under the section for screenshots?

Other than that and the comments I left about adding the new line at the end of the file. It looks good. Just make those changes and I can approve.

File: src/pages/index.js [hunk 5]
```diff
@@ -15,109 +43,10 @@ export default function Home() {
...
-            <h2 className={inter.className}>
-              Deploy <span>-&gt;</span>
-            </h2>
-            <p className={inter.className}>
-              Instantly deploy your Next.js site to a shareable URL
-              with&nbsp;Vercel.
-            </p>
-          </a>
-        </div>
+        {listings.map((data) => (
```
Comment (annonymousauthority) [inline src/pages/index.js:46]: Did this actually render?

Comment (ibimina) [reply src/pages/index.js:46]: > Did this actually render?

yes, it did

File: src/styles/globals.css (hunk 2 above)
Comment (ibimina) [reply src/styles/globals.css:14]: ok

File: src/styles/Home.module.css (hunk 3 above)
Comment (ibimina) [reply src/styles/Home.module.css:17]: ok

File: src/components/landing/listing.module.css (hunk 4 above)
Comment (ibimina) [reply src/components/landing/listing.module.css:12]: ok

File: src/components/landing/listing.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+          <Image src="" width={50} height={50} alt={`${data.name} avatar`} />
```
[the remark this answers is not in the document]
Comment (ibimina) [reply src/components/landing/listing.js]: okay

File: src/components/landing/listing.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
+import Link from "next/link";
+import Image from "next/image";
+import styles from "./listing.module.css";
```
Comment (msarit) [inline src/components/landing/listing.js:3]: This is the CSS approach that I am learning in my React course 😅  I like the approach.

**Review by jjleigh — APPROVED** [review]
Looks good! 

When you add screenshots to the PR description ensure that the formatting isn't disturbed. Not an issue but I just wanted to point out that the formatting for the text under the screenshot got a bit messed up.

**Comment by ibimina** [comment]
> Looks good!
> 
> When you add screenshots to the PR description ensure that the formatting isn't disturbed. Not an issue but I just wanted to point out that the formatting for the text under the screenshot got a bit messed up.

Corrected

File: src/styles/Home.module.css [hunk 8]
```diff
@@ -1,278 +1,18 @@
...
-  to {
-    transform: rotate(0deg);
-  }
-}
+@media (min-width:60rem) {
+    .listings_container {
+        grid-template-columns: repeat(3, 1fr);
+        grid-gap: 1em;
+    }
+}
```
Comment (jjleigh) [inline src/styles/Home.module.css:18]: Add a new line at the end here

**Comment by msarit** [comment]
@ibimina You need to resolves conflicts my dear before I review; I am reviewing on my local app, not just on github

**Comment by ibimina** [comment]
alright will do that

**Comment by ibimina** [comment]
@msarit I have resolved the conflicts

File: src/components/landing/listing.js [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+function Listing({ data }) {
+  return (
+    <>
+      <article className={styles.listing}>
+        <div className={styles.listing__image__container}>
+          <Image src="/image-placeholder.png" width={50} height={50} alt={`${data.name} avatar`} />
+        </div>
+        <div className={styles.listing__content}>
+          <h2 className={styles.listing__content__name}>{data.community_name}</h2>
+          <p className={styles.listing__content__manager}>{data.manager}</p>
```
Comment (annonymousauthority) [inline src/components/landing/listing.js:13]: I think this should have a different styling from the `data.description` paragraph, just to improve UX and follow basic design principle

**Review by annonymousauthority — APPROVED** [review]

File: src/components/landing/listing.js (hunk 9 above)
Comment (ibimina) [reply src/components/landing/listing.js:13]: fixed @annonymousauthority

**Comment by ibimina** [comment]
@msarit @jjleigh @annonymousauthority  I have resolved conflicts

**Comment by annonymousauthority** [comment]
I think this is ready to be merged @msarit

**Pull request merged**